### PR TITLE
Sync script for translation repos

### PR DESCRIPTION
**Pull request by tesseralis**
## Description

Create a sync script that updates translation repositories based on the current state of the source repo.

Usage:

```
yarn run sync [language code]
```

## Behavior

1. Creates a new PR that merges the source repo `gatsby-i18n-source` to the current head of the translation repo, committing all merge conflicts
2. Adds comments referring to the changes in the source between the last sync and now, to help translators decide what content needs to be updated.

## Next steps

* [ ] Don't create a PR if there is an existing one
* [ ] Watch source repo and run on all translation repos when there are updates
* [x] Automatically merge the PR if there are no conflicts
* [ ] Documentation for how to resolve a sync script (pending #21254)
* [x] Decide if we want to commit non-conflicting files separately

## Examples

* Japanese: https://github.com/gatsbyjs/gatsby-ja/pull/155
* Hindi: Sync: https://github.com/gatsbyjs/gatsby-hi/pull/44 / Conflicts: https://github.com/gatsbyjs/gatsby-hi/pull/43

## Documentation

#21753 

## Related issues

Fixes: #21250

### CODE REVIEW

File: www/src/assets/vendor-logos/codesandbox.svg [hunk 1]
```diff
@@ -1,4 +1,4 @@
...
+<svg width="23" height="23" viewBox="0 0 23 23" fill="none" xmlns="http://www.w3.org/2000/svg" focusable="false">
```
Comment (tesseralis) [inline www/src/assets/vendor-logos/codesandbox.svg]: ...I dunno why the linter decided to pick this up on this PR.

File: docs/contributing/translation/sync-guide.md [hunk 2]
```diff
@@ -1,13 +1,32 @@
 ---
 title: Keeping Translations Up-to-date
-issue: https://github.com/gatsbyjs/gatsby/issues/21250
 ---
 
-Periodically, gatsbybot will create pull requests to keep translations repos up-to-date with the original English source. Make sure to review these PRs to ensure that your translation remains accurate.
+Periodically, gatsbybot will update your translation repo to be up-to-date with the current English repo. If there is an update to a page that is already translated, gatsbybot will create a pull request listing the conflicts between translation and the new English content. Resolving these conflicts and merging these pull requests is essential to keeping your translation repo up-to-date.
 
-> Note: the bot doesn't work yet but will come soon. Until then, see the next section to learn how to manually sync your repo.
+## Resolving sync pull requests
```
Comment (LekoArts) [inline docs/contributing/translation/sync-guide.md]: Should we add a TODO or open an issue to update that part with one real-world example (e.g. spanish/japanese repo) that shows those two PRs and how they were merged (in best practice)?

File: scripts/i18n/sync.js [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+        authorization: `token ${process.env.GITHUB_BOT_AUTH_TOKEN}`,
+        accept: `application/vnd.github.shadow-cat-preview+json`,
+      },
+      input,
+    }
+  )
+  return createPullRequest.pullRequest
+}
+
+function conflictPRBody(conflictFiles, comparisonUrl, prNumber) {
```
Comment (LekoArts) [inline scripts/i18n/sync.js]: Let's also link to our docs here at the bottom 👍

**Comment by AishaBlake** [comment]
@tesseralis this looks good and I've seen it work. I've also read through the WIP docs in #21753. Do you intend to work on any more of the "next steps" in this PR or do you feel comfortable moving forward? I think I'd prefer to merge with the functionality this has now and then work in smaller chunks to tackle additional work.

**Comment by tesseralis** [comment]
@AishaBlake I'd like to merge this in now and work on the "next steps" in a different PR.

**Comment by AishaBlake** [comment]
Should probably also check in with @LekoArts. Lennart, feel free to merge if you're good with this! I'd also like to get #21753 in soon after.

**Comment by tesseralis** [comment]
@LekoArts it's only starters validate, it's failing because of npm audits.

**Pull request merged**